api/v1alpha1/vaultdynamicsecret_types.go

```diff
@@ -46,6 +46,11 @@ type VaultDynamicSecretSpec struct {
 	RenewalPercent int `json:"renewalPercent,omitempty"`
 	// Revoke the existing lease on VDS resource deletion.
 	Revoke bool `json:"revoke,omitempty"`
+	// StaticCreds should be set when syncing credentials that are periodically
+	// rotated by the Vault server, rather than created upon request. These secrets
+	// are sometimes referred to as "static roles", or "static credentials", with a
+	// request path that contains "static-creds".
+	StaticCreds bool `json:"staticCreds,omitempty"`
```

Comment:
Would it be possible to determine if the creds are static just by inspecting the Vault response?

Reply:
I was thinking this as well.
It looks like for DB there isnt anything in the (sample) response in the api-docs that screams out as "static-cred". We can parse the path `/database/static-creds/`, but that doesn't seem future-proof.

Reply:
Yeah, that might be possible. I hesitated on that originally, since I was not sure that we could guarantee that response is really for "static-creds". It's possible we could come up with some heuristic that would work most of the time, but we would have to provide an escape hatch for the user in case VSO gets it wrong.

We could do something like:
```
isStaticCreds = leaseNotRenewable && some-static-creds-keys in resp.Data
```

I sort of prefer the more declarative approach that we have now, but I could definitely be persuaded otherwise.

Reply:
I think there is definitely enough in the response to figure this out (at the time of getting a credential, which may be too late in the logic of Reconcile):

Example response for dynamic:
```
{
  "data": {
    "username": "root-1430158508-126",
    "password": "132ae3ef-5a64-7499-351e-bfe59f3a2a21"
  }
}
```

and static:
```
{
  "data": {
    "username": "static-user",
    "password": "132ae3ef-5a64-7499-351e-bfe59f3a2a21",
    "last_vault_rotation": "2019-05-06T15:26:42.525302-05:00",
    "rotation_period": 30,
    "ttl": 28
  }
}
```

But I do agree that it feels hacky to rely on the response alone, and a more declarative approach feels cleaner and might make the code easier to read.

Reply:
Updated with 41e9143257679ed973a6d9ecf9f7fbe68bc41494